src/main/java/org/jboss/aesh/cl/populator/AeshCommandPopulator.java

```diff
@@ -70,6 +80,35 @@ else if(option.getDefaultValues().size() > 0) {
         }
         else if(line.getArgument() != null)
             resetField(getObject(), line.getArgument().getFieldName(), true);
+
+		if (commandInvocation != null) {
```

Comment:
This is the biggest question for me.  `populateObject` is called twice: once as arguments and options are being completed (`AeshCommandLineCompletionParser`), and again during command execution (`DefaultCommandContainer`).  Realistically, input prompts are handled only by the latter, since it has the `CommandInvocation` available.

Rather than have `AeshCommandLineCompletionParser` pass `null` for that new `commandInvocation`, I'm wondering if this would be better to have split into multiple methods, specific to completion and execution.

Admittedly, I'm pretty confused as to why `AeshCommandLineCompletionParser` calls this any way -- couldn't really figure it out with debugging.  Shouldn't the command's fields be set only when it's actually executed?  If so, couldn't this call be delayed and handled purely by `DefaultCommandContainer`?